### ISSUE TITLE
Better Input Checking for Minimum Score

### ISSUE THREAD

**Issue by DarioS** [BenLangmead/bowtie2#64]
I used `--score-min L,18 --end-to-end` but the software didn't stop with an error that the maximum possible score when `--end-to-end` is used is 0. This could be improved.

**Comment by BenLangmead** [comment]
Agree.  We have a little bit of checking like that in 2.3.0 and later, but it could be expanded.  We'll have to be careful to achieve the right level of pickiness.  For example, right now we complain about `--local --score-min G,0,10` when we do this check:

https://github.com/BenLangmead/bowtie2/blob/master/bt2_search.cpp#L1574

But that's probably being too aggressive.
